WebMIDI security and permission API

hamishwillee opened the pull request:
To access webMidi you need to call `requestMIDIAccess()` in a secure context, the page permission policy must allow access, and the user must have granted access. The permission status can be queried using the permission API.

I have added a "Security requirements" section in both the Web MIDI API overview and in `requestMIDIAccess()` to capture this (permissions API behaviour is defined in the Web MIDI API spec so this is a reasonable place). This matches the pattern set by @Elchi3 previously.

EDITED: This description now says what I did. Previously was request to @Elchi3 @wbamberg for their thoughts.

Note, this is part of #23680

hamishwillee committed:
requestMidiAccess - security consideration

hamishwillee committed:
Web MIDI API - add security considerations

hamishwillee committed:
Add compatibility tables to web API

hamishwillee commented on files/en-us/web/api/web_midi_api/index.md:
```diff
@@ -107,6 +134,14 @@ function startLoggingMIDIInput(midiAccess, indexOfPort) {
 
+### Permission API
+
+{{Compat("api.Permissions.permission_midi")}}
```
Logically this is probably the right place for this since it is part of the midi api spec and separate from actually granting access.

Elchi3 commented:
This looks great to me! The only difference to what I did previously is that I called these sections "Security requirements" (see e.g. https://developer.mozilla.org/en-US/docs/Web/API/Performance/measureUserAgentSpecificMemory#security_requirements).

hamishwillee commented:
@Elchi3 Thanks - but would you include security info in one place or both (top level overview or the place where it is first tested?).

I'm starting to lean towards "both", even though that means duplication.

Elchi3 commented:
I think both is fine. I guess you really need to know that this is a prerequisite when using the API, so I think it makes sense to show it in both places.

hamishwillee committed:
Merge branch 'main' into webapi_security

hamishwillee committed:
Security considerations - make identical

hamishwillee committed:
Rebase fix

hamishwillee commented on files/en-us/web/api/web_midi_api/index.md:
```diff
@@ -9,7 +9,10 @@ tags:
+browser-compat:
+  - api.Navigator.requestMIDIAccess
+  - http.headers.Permissions-Policy.midi
+  - api.Permissions.permission_midi
```
@queengooborg `api.Permissions.permission_midi` finds the specification but does not update the compat tables. I have tested and the other permissions are not found either, though  `api.Permissions` itself does update the compat table. Any thoughts on what might cause this?

hamishwillee commented:
OK, this is now ready for review. 

FYI @Elchi3 Since you saw this last I updated to match your heading name convention. I also made the security in the overview and the method almost exactly the same, and moved the permission query example as part of the security section (rather than as a standalone example).

hamishwillee commented on files/en-us/web/api/web_midi_api/index.md:
```diff
@@ -38,6 +41,28 @@ Therefore, the API can be used for musical and non-musical uses, with any MIDI d
 - {{domxref("MIDIConnectionEvent")}}
   - : The event passed to the {{domxref("MIDIAccess.statechange_event")}} and {{domxref("MIDIPort.statechange_event")}} events, when a port becomes available or unavailable.
 
+## Security requirements
```
@queengooborg Further, the `api.Permissions.permission_midi` does not have an MDN link - we just have one for all the Permissions. We don't have specific MDN pages for permission API keys such as `midi`, though we do for permission policy HTTP keys. 

I would _like" to link to **this** section from the BCD. It covers the use of the permission status `midi`, and is currently the best place for a link.

Does this make sense to you?

hamishwillee commented:
@Elchi3 @sideshowbarker Would one of you be able to approve this?

hamishwillee committed:
Merge branch 'main' into webapi_security
